### DISCUSSION TITLE
Run only in user context or separate black-/whitelist for user/system context

### DISCUSSION THREAD

**Discussion by arjangit1234** [Romanitho/Winget-AutoUpdate#661] [Ideas]
We've implemented Winget-AutoUpdate through Intune using ADMX-templates. 
For applications installed in system context we'll provide updates through Intune. However we would like to update applications installed in the user-context. 
Right now we've managed to do so by setting up a whitelist and using the option to ignore the black/whitelist in user context. This way nothing in the system contexts gets updated, but everything in the user context will be updated. However this means we're not able to use the black or whitelist anymore for what they're designed to do. It would therefore be nice to have separate black/whitelist for user and system context or at least an option to not install any updates in the system context.

**Comment by KnifMelti** [comment]
Smart move! 👌
Most applications installed under the user context update themselves upon starting. Are you concerned about outdated files in the user context from applications that have not been started for a long time?

**Reply by Romanitho** [reply]
Maybe we can remove all the triggers on the system task and put the triggers on the user context task? Not tested, but i could work I guess.

**Comment by arjangit1234** [comment]
That's indeed the reason we're using it this way. A lot of software automatically updates but not everything. We want to have all software up to date even it's not used at the moment. This way we make sure the software is already up to date when it's started again after a long time. Also this way we have a better overview of outdated software in Microsoft Defender. 

I've tried to disable the Winget-AutoUpdate task and add triggers to the Winget-AutoUpdate-UserContext. Now it runs only in the user context so that would work. Is there any way to set this up through a policy? Right now we're only testing with a small amount of devices but eventually we'd like to deploy it to 700+ devices.

**Reply by Romanitho** [reply]
There is no way to handle that by policies. It was not in the pipe to be honest, because from our point of view, there is no reason to get apps installed in the user context (except few). Moreover, it can happen that WAU updates user apps but require admin right to achieve the action.
At the moment we are quite focus on building an MSI version. It takes time, need lot of tests, etc... but then, we can try to work on your case to see if we can control the user context only :)

**Reply by arjangit1234** [reply]
Thanks for your reply! There is indeed no reason to install centrally managed apps in the user context. However users themselves install software in the user context. Although that is not the preferred installation method we would like to keep also that software up to date as much as possible and I think WAU would be a nice solution for this case. You're right about some software requiring admin rights, that's why I'd like to blacklist some apps (e.g. Dev Home) or use whitelisting to update specific apps.  

If you have some time to have a look, after the MSI is finished, that would be nice!

**Reply by Romanitho** [reply]
sure, we will try to see how it can be handled. But as you can guess, it is not our top priority atm 😅
